### ISSUE TITLE
ResolveTargetingPackAssets crash in Dev16

### ISSUE THREAD

**Issue by nguerrera** [dotnet/sdk#2940]
@DamianEdwards reports the stack below in error list with latest SDK, latest Dev16 and the following project created from template:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>


  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview-19106-07" />
  </ItemGroup>

</Project>
```

![image](https://user-images.githubusercontent.com/75470/52815895-b5621180-3054-11e9-9d0f-dd70c23d59b5.png)

cc @dsplaisted @natemcmaster

**Comment by nguerrera** [comment]
From the error, it looks like the targeting pack path may have resolved to empty and we went looking relative to project path. I suspect something about the design-time build case where there has been some trickiness.

This is consistent with the report that it worked from CLI.

**Comment by nguerrera** [comment]
CLI version: 3.0.100-preview4-010331

This could be missing RI from preview3 to preview4 as there has been significant change to these targets in preview3 and not sure if we've RIed them to preview4.

**Comment by nguerrera** [comment]
REpros with latest preview3. Now trying latest code in this repo. It goes away for me once restore succeeds. This is almost certainly a bug in the handling of the pre-restore design time build.

**Comment by nguerrera** [comment]
There's a race here, repro is consistent with:

* Close VS
* `dotnet nuget locals all --clear` 
* Reopen VS
* Create new ASP.NET Core Web Application

**Comment by nguerrera** [comment]
This does not repro with latest bits of release/3.0.1xx of this repo. This hasn't made it to a core-sdk build yet. There was a subscription error that @mmitche is fixing in the 3.0 flow. And master is out of date with the changes that fixed it. We should have a preview3 build with the fix tomorrow. 

@peterhuene is looking at automating preview3 -> master (preview4) so that should follow sooner or later too.

So basically, this is fixed, but we have code flow delays.

I'll leave this open and check for a known good build for folks to try before closing.

**Comment by jimmylewis** [comment]
I'm running into this too, with the released preview 2 bits.  Can you point me to a newer build that I can verify it's fixed in?

My failing scenario is during build on CI (VS2019 Preview 3 Build Tools SKU, .NET Core 3 Preview 2 SDK).

**Comment by dsplaisted** [comment]
@jimmylewis the latest daily build from the Release/3.0.1XX branch linked [here](https://github.com/dotnet/core-sdk) should have the fix.

However, we believe this specific bug was specific to Visual Studio.  So if you still see a failure with the latest build, I'd recommend opening a new issue.

**Comment by jimmylewis** [comment]
Thanks @dsplaisted.  I'm running into a different error now, so I'll open a new issue to track that.

**Comment by springy76** [comment]
My solution builds fine on several PCs where VS2019-Preview4 or VS2019-RC is installed, but fails with that exact ResolveTargetingPackAssets error (and the non-existant project\ref\... path)  on the build server where only VS2019-BuildTools are installed. The netcore3-SDK everywhere has to same version.
What shall I change on the build server to get it working?

**Comment by jimmylewis** [comment]
@springy76 Updating to Preview 3 fixed it for me.  I ran into an issue with Razor Components going from preview 2 to preview 3 (#2977).  After updating my project I didn't encounter any other issues.

**Comment by springy76** [comment]
I installed Build Tools Preview4 and Build Tools RC, too: Both time the same problem.

Even tried a daily netcore SDK build, which suprised me with even new errors.

**Comment by nguerrera** [comment]
@springy76 We'll take a look, can you open two new bugs:

1. For the build tools + .NET Core Preview 3
2. For the new errors in .NET Core Preview 4 daily build

For the first one, can you gather a [Binary Log](https://gist.github.com/dsplaisted/b256d7804672a6f6375476a5f5658b7b)? Make note of the fact that this will capture all environment variables (possibly with secrets in your build and project file contents before generating or sharing.

**Comment by springy76** [comment]
I didn't recognize that .NET Core Preview 3 is out now (since some hours).

The new error with that is now:
(_CopyOutOfDateSourceItemsToOutputDirectory target) ->  c:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(4560,5): error MSB3030: Could not copy the file "PathToMyProject\obj\Release\netcoreapp3.0\MyProject.exe" because it was not found.

**Comment by nguerrera** [comment]
@springy76 Please file a new bug with repro steps or [Binary Log](https://gist.github.com/dsplaisted/b256d7804672a6f6375476a5f5658b7b)

**Comment by springy76** [comment]
@nguerrera I found the problem: It was the (our) specific build server which restored Nuget packages using `nuget.exe restore`. I changed this to `msbuild -t:restore` and now everything is fine.
